pyccel/codegen/printing/ccode.py

```diff
@@ -1134,35 +1182,52 @@ def formatted_args_to_printf(args_format, args, end):
             code += formatted_args_to_printf(args_format, args, end)
         return code
 
-    def find_in_dtype_registry(self, dtype):
+    def get_c_type(self, dtype):
         """
-        Find the corresponding C dtype in the dtype_registry.
+        Find the corresponding C type of the PyccelType.
 
-        Find the corresponding C dtype in the dtype_registry.
-        Raise PYCCEL_RESTRICTION_TODO if not found.
+        For scalar types, this function searches for the corresponding C data type
+        in the `dtype_registry`.  If the provided type is a container (like
+        `HomogeneousSetType` or `HomogeneousListType`),  it recursively identifies
+        the type of an element of the container and uses it to calculate the
+        appropriate type for the `STC` container.
+        A `PYCCEL_RESTRICTION_TODO` error is raised if the dtype is not found in the registry.
 
         Parameters
         ----------
-        dtype : DataType
-            The data type of the expression.
+        dtype : PyccelType
+            The data type of the expression. This can be a fixed-size numeric type,
+            a primitive type, or a container type.
 
         Returns
         -------
         str
-            The code which declares the datatype in C.
+            The code which declares the data type in C or the corresponding `STC` container
+            type.
+
+        Raises
+        ------
+        PYCCEL_RESTRICTION_TODO
+            If the dtype is not found in the dtype_registry.
```

Comment:
so `PyccelCodegenError` instead of `PYCCEL_RESTRICTION_TODO`

Reply:
updated c381e60cad0b9cc554ac36c1a170f2b796f011e5

Reply:
Yes sorry. This type is useful because it tells you that you can do:
```python
try:
    ctype = self.get_c_type(my_type)
except PyccelCodegenError:
    ctype = 'void*'
```